samples/csharp/getting-started/Ranking_AutoML/Ranking/Ranking/Program.cs

```diff
@@ -0,0 +1,234 @@
+            var firstGroupId = searchQueries.First().GroupId;
+            var firstGroupPredictions = searchQueries.Take(100).Where(p => p.GroupId == firstGroupId).OrderByDescending(p => p.Score).ToList();
+
+            // The individual scores themselves are NOT a useful measure of result quality; instead, they are only useful as a relative measure to other scores in the group. 
+            // The scores are used to determine the ranking where a higher score indicates a higher ranking versus another candidate result.
+            foreach (var prediction in firstGroupPredictions)
+            {
+                Console.WriteLine($"GroupId: {prediction.GroupId}, Score: {prediction.Score}");
+            }
+        }
+
+        private static void GetData(string inputPath, string outputPath, string trainDatasetPath, string trainDatasetUrl,
+            string testDatasetUrl, string testDatasetPath, string validationDatasetUrl, string validationDatasetPath)
+        {
+            Console.WriteLine("===== Prepare data =====\n");
+
+            if (!Directory.Exists(outputPath))
+            {
+                Directory.CreateDirectory(outputPath);
+            }
+
+            if (!Directory.Exists(inputPath))
+            {
+                Directory.CreateDirectory(inputPath);
+            }
+
+            if (!File.Exists(trainDatasetPath))
+            {
+                Console.WriteLine("===== Downloading the train dataset - this may take several minutes =====\n");
+                using (var client = new WebClient())
+                {
+                    client.DownloadFile(trainDatasetUrl, TrainDatasetPath);
+                }
+            }
+
+            if (!File.Exists(validationDatasetPath))
+            {
+                Console.WriteLine("===== Downloading the validation dataset - this may take several minutes =====\n");
+                using (var client = new WebClient())
+                {
+                    client.DownloadFile(validationDatasetUrl, validationDatasetPath);
+                }
+            }
+
+            if (!File.Exists(testDatasetPath))
+            {
+                Console.WriteLine("===== Downloading the test dataset - this may take several minutes =====\n");
+                using (var client = new WebClient())
+                {
+                    client.DownloadFile(testDatasetUrl, testDatasetPath);
+                }
+            }
+
+            Console.WriteLine("===== Download is finished =====\n");
+        }
+
+        private static void PrintTopModels(ExperimentResult<RankingMetrics> experimentResult)
+        {
+            // Get top few runs ranked by nDCG
+            var topRuns = experimentResult.RunDetails
+                .Where(r => r.ValidationMetrics != null && !double.IsNaN(r.ValidationMetrics.NormalizedDiscountedCumulativeGains[0]))
+                .OrderByDescending(r => r.ValidationMetrics.NormalizedDiscountedCumulativeGains[9]).Take(5);
```

Comment:
I'm concerned users will modify `OptimizationMetricTruncationLevel` above, and being hard hard coded here, will become out of sync. 

The sorting here is different than in other places, which also hard code `[9]`. In other places, it's used for display purposes only and we call out that it's displaying `NDCG@10`. Here, the `experimentResult.BestRun` may not match `topRuns[0]` if the user modifies `OptimizationMetricTruncationLevel`.

Do you see a good way to access, or pass along, the `OptimizationMetricTruncationLevel`? I don't think there's a way to access the `RankingExperimentSettings` from `experimentResult`. So perhaps it's easier to just add a parameter to `PrintTopModels()`:
```diff
- private static void PrintTopModels(ExperimentResult<RankingMetrics> experimentResult)
+ private static void PrintTopModels(ExperimentResult<RankingMetrics> experimentResult, uint optimizationMetricTruncationLevel)
```
<br/>

And of course use here:
```suggestion
                .OrderByDescending(r => r.ValidationMetrics.NormalizedDiscountedCumulativeGains[optimizationMetricTruncationLevel - 1]).Take(5);
```
<br/>

And list the specific `@N`: _(and changing wording from "ranked" to "sorted")_
```diff
- Console.WriteLine("Top models ranked by nDCG --");
+ Console.WriteLine($"Top models sorted by nDCG@{optimizationMetricTruncationLevel} --");
```

Reply:
#Resolved